Assets/MRTK/Examples/Demos/UX/ProgressIndicator/Scripts/ProgressIndicatorDemo.cs

```diff
@@ -0,0 +1,138 @@
+    /// </summary>
+    public class ProgressIndicatorDemo : MonoBehaviour
+    {
+        [SerializeField, Header("Indicators")]
+        private GameObject progressIndicatorLoadingBarGo = null;
+        [SerializeField]
+        private GameObject progressIndicatorRotatingObjectGo = null;
+        [SerializeField]
+        private GameObject progressIndicatorRotatingOrbsGo = null;
+
+        [SerializeField, Header("Editor Keyboard Controls")]
+        private KeyCode toggleBarKey = KeyCode.Alpha1;
+        [SerializeField]
+        private KeyCode toggleRotatingKey = KeyCode.Alpha2;
+        [SerializeField]
+        private KeyCode toggleOrbsKey = KeyCode.Alpha3;
+
+        [SerializeField, Header("Settings")]
+        private string[] loadingMessages = new string[] { 
+            "First Loading Message",
+            "Loading Message 1", 
+            "Loading Message 2", 
+            "Loading Message 3", 
+            "Final Loading Message" };
+
+        [SerializeField, Range(1f, 10f)]
+        private float loadingTime = 5f;
+
+        private IProgressIndicator progressIndicatorLoadingBar;
+        private IProgressIndicator progressIndicatorRotatingObject;
+        private IProgressIndicator progressIndicatorRotatingOrbs;
+
+        /// <summary>
+        /// Target method for demo button
+        /// </summary>
+        public void OnClickBar()
+        {
+            HandleButtonClick(progressIndicatorLoadingBar);
+        }
+
+        /// <summary>
+        /// Target method for demo button
+        /// </summary>
+        public void OnClickRotating()
+        {
+            HandleButtonClick(progressIndicatorRotatingObject);
+        }
+
+        /// <summary>
+        /// Target method for demo button
+        /// </summary>
+        public void OnClickOrbs()
+        {
+            HandleButtonClick(progressIndicatorRotatingOrbs);
+        }
+
+        private async void HandleButtonClick(IProgressIndicator indicator)
+        {
+            // If the indicator is opening or closing, wait for that to finish before trying to open / close it
+            // Otherwise the indicator will display an error and take no action
+            await indicator.AwaitTransitionAsync();
+
+            switch (indicator.State)
+            {
+                case ProgressIndicatorState.Closed:
+                    OpenProgressIndicator(indicator);
+                    break;
+
+                case ProgressIndicatorState.Open:
+                    await indicator.CloseAsync();
+                    break;
+            }
+        }
+
+        private void OnEnable()
+        {
+            progressIndicatorLoadingBar = progressIndicatorLoadingBarGo.GetComponent<IProgressIndicator>();
+            progressIndicatorRotatingObject = progressIndicatorRotatingObjectGo.GetComponent<IProgressIndicator>();
+            progressIndicatorRotatingOrbs = progressIndicatorRotatingOrbsGo.GetComponent<IProgressIndicator>();
+        }
```

Comment:
Out of curiosity, is there a special reason this is in OnEnable vs Start/Awake?

Reply:
Arbitrary choice.